source/portable/NetworkInterface/NXP1060/NetworkInterface.c

```diff
@@ -0,0 +1,666 @@
+            ethernetifLocal->base = ( void * ) ENET_BASE;
+
+            /* Set the clock frequency. */
+            mdioHandle.resource.csrClock_Hz = EXAMPLE_CLOCK_FREQ;
+            mdioHandle.resource.base = ( void * ) ENET_BASE;
+
+            /* Set the configuration to auto-negotiate; set the phy to full duplex mode; the phy's address; and the speed. */
+            xConfig.autoNeg = pdTRUE;
+            xConfig.duplex = kPHY_FullDuplex;
+            xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+            xConfig.speed = kPHY_Speed100M;
+
+            FreeRTOS_printf( ( "Starting PHY initialization.\r\n" ) );
+
+            xStatus = PHY_Init( &phyHandle, &xConfig );
+
+            if( xStatus != kStatus_Success )
+            {
+                FreeRTOS_printf( ( "Failed to initialize the PHY..." ) );
+                break;
+            }
+
+            eEMACState = xEMAC_WaitPHY;
+        /* Fall through. */
+
+        case xEMAC_WaitPHY:
+            FreeRTOS_printf( ( "Configuration successful. Waiting for auto-negotiation to complete...\r\n" ) );
+
+            do
+            {
+                readStatus = PHY_GetAutoNegotiationStatus( &phyHandle, &autoNegotiationComplete );
+
+                if( autoNegotiationComplete == true )
+                {
+                    break;
+                }
+
+                /* Try for only a limited number of times. */
+                if( ucCounter++ > MAX_AUTONEG_FAILURE_COUNT )
+                {
+                    break;
+                }
+
+                vTaskDelay( pdMS_TO_TICKS( 10 ) );
+            }
+            while( readStatus == kStatus_Success );
+
+            if( autoNegotiationComplete == false )
+            {
+                FreeRTOS_printf( ( "Failed to complete auto-negotiation. Using default values." ) );
+
+                xConfig.autoNeg = pdFALSE;
+                xConfig.duplex = kPHY_FullDuplex;
+                xConfig.phyAddr = BOARD_ENET0_PHY_ADDRESS;
+                xConfig.speed = kPHY_Speed100M;
+
+                xStatus = PHY_Init( &phyHandle, &xConfig );
+
+                if( xStatus != kStatus_Success )
+                {
+                    FreeRTOS_printf( ( "Failed to re-initialize the PHY with default configs too..." ) );
+                    eEMACState = xEMAC_Fatal;
+                    break;
+                }
+                else
+                {
+                    FreeRTOS_printf( ( "PHY re-initialization successful." ) );
+                }
+            }
+            else
+            {
+                FreeRTOS_printf( ( "Auto-negotiation complete.\r\n" ) );
+            }
+
+            ucCounter = 0;
+
+            do
+            {
+                readStatus = PHY_GetLinkStatus( &phyHandle, &LinkUp );
+
+                if( LinkUp == true )
+                {
+                    break;
+                }
+
+                vTaskDelay( pdMS_TO_TICKS( 1000 ) );
```

Comment:
Make it a bit shorter than a whole second? Maybe 50 ms and doing more loops?

Reply:
Fixed too